pkg/runtime/serializer/codec_factory.go

```diff
@@ -308,3 +310,52 @@ func (f CodecFactory) SerializerForFileExtension(extension string) (runtime.Seri
 	}
 	return nil, false
 }
+
+type DirectCodecFactory struct {
```

Comment:
@smarterclayton these are the important bits of this PR. Others are mostly generated code.